yt/visualization/plot_window.py

```diff
@@ -665,6 +667,9 @@ def __init__(self, *args, **kwargs):
         self._splat_color = kwargs.pop("splat_color", None)
         PlotWindow.__init__(self, *args, **kwargs)
 
+    def data_is_valid(self):
+        return self._data_valid and self._frb and self._frb._data_valid
```

Comment:
Now that you've refactored everything into this function, do you think we need both `_data_valid` and `frb._data_valid`? Can't we just use `frb._data_valid` and get rid of the attribute on the plot itself?

I also like to make helpers like this properties so you don't need to append the `()` every time you use them.